### PR TITLE
Makes anonymous donation field false by default

### PR DESCRIPTION
**Pull request by clarissalimab**
To fix the existing data, we need to run:

```ruby
Donation.where(anonymous: nil).find_in_batches do |dons| 
  ids = dons.map(&:id) 
  Donation.where("id IN (?)", ids).update_all(anonymous:false, updated_at: Time.current) 
end
```

Part of [tix #3259](https://github.com/CommitChange/tix/issues/3259).

### CODE REVIEW

File: db/migrate/20210930192755_change_anonymous_default_from_donations.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ChangeAnonymousDefaultFromDonations < ActiveRecord::Migration
+  def change
+    change_column_default :donations, :anonymous, false
+    change_column_null :donations, :anonymous, false
```
Comment (clarissalimab) [inline db/migrate/20210930192755_change_anonymous_default_from_donations.rb]: Do you think it would be better to change this in a separate PR after we merge the first part?

File: app/models/donation.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
 class Donation < ActiveRecord::Base
 
+	before_save :set_anonymous
+
```
Comment (wwahammy) [inline app/models/donation.rb]: This isn't needed. If we deploy the change to to set the default to false, there's no time when a donation will be created with anonymous as nil.

File: db/migrate/20210930192755_change_anonymous_default_from_donations.rb (hunk 1 above)
Comment (wwahammy) [inline db/migrate/20210930192755_change_anonymous_default_from_donations.rb]: This line shouldn't happen yet as we need to do this in three steps.

**Review by wwahammy — CHANGES_REQUESTED** [review]
I think your solution is more complex and brittle than it needs to be.

Three steps described in the referenced issue don't need to happen all at once and I specifically wrote them in a particular order.

File: db/migrate/20210930192755_change_anonymous_default_from_donations.rb (hunk 1 above)
Comment (wwahammy) [reply db/migrate/20210930192755_change_anonymous_default_from_donations.rb]: I do! Merge the first part and update prod run the script to correct all the old ones and merge this and update prod again. Honestly, I'm not sure we even need this line as the first line should fix it.

File: db/structure.sql [hunk 3]
```diff
@@ -792,7 +792,7 @@ CREATE TABLE public.donations (
     card_id integer,
     designation text,
     offsite boolean,
-    anonymous boolean,
+    anonymous boolean DEFAULT false NOT NULL,
```
Comment (wwahammy) [inline db/structure.sql]: I think you might have an old migration still in place! This shouldn't be `NOT NULL.` To cleanup, clear your db,  and update from a recent database.

`bin/rake db:drop db:create`
`script/restore_from_heroku.sh` or `script/pg_restore_local_from_production.sh`

and then run `bin/rake db:migrate`

File: db/structure.sql [hunk 4]
```diff
@@ -2552,7 +2552,7 @@ CREATE TABLE public.supporters (
     address character varying(255),
     city character varying(255),
     state_code character varying(255),
-    anonymous boolean,
+    anonymous boolean DEFAULT false,
```
Comment (wwahammy) [inline db/structure.sql]: Also from a migration in progress.

File: db/structure.sql [hunk 5]
```diff
@@ -5725,3 +5725,7 @@ INSERT INTO schema_migrations (version) VALUES ('20210818221411');
 
 INSERT INTO schema_migrations (version) VALUES ('20210917204956');
 
+INSERT INTO schema_migrations (version) VALUES ('20210930192755');
+
+INSERT INTO schema_migrations (version) VALUES ('20210930194624');
```
Comment (wwahammy) [inline db/structure.sql]: Same thing, there should only be one new migration

**Review by wwahammy — CHANGES_REQUESTED** [review]
Nice work. It's exactly right except the structure.sql is a little wonky.

**Comment by clarissalimab** [comment]
I ran the tests for this one with the current changes.

**Review by wwahammy — APPROVED** [review]
Looks good. :rocket:

**Comment by clarissalimab** [comment]
@wwahammy I'm going to merge it, deploy it, run the migration and run the script, okay?

**Comment by wwahammy** [comment]
Sounds good!
